winlogbeat/sys/strings_windows.go

```diff
@@ -0,0 +1,48 @@
+// Licensed to Elasticsearch B.V. under one or more contributor
+// license agreements. See the NOTICE file distributed with
+// this work for additional information regarding copyright
+// ownership. Elasticsearch B.V. licenses this file to you under
+// the Apache License, Version 2.0 (the "License"); you may
+// not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+package sys
+
+import (
+	"golang.org/x/sys/windows"
+	"golang.org/x/text/encoding"
+	"golang.org/x/text/encoding/charmap"
+)
+
+var ansiDecoder *encoding.Decoder
+
+func init() {
```

Comment:
What is the impact of rewriting this to avoid performing any work in init, instead using lazy initialization upon first use via something like sync.OnceValue?

I've had a few bad experiences with packages doing work in init, as it can be hard to debug when issues arise. Additionally, any program that imports this package will incur the init cost, even if it’s not used.